code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm

```diff
@@ -17,12 +17,49 @@
 	inherent_verbs = list(
 		/mob/living/carbon/xenomorph/proc/vent_crawl,
 	)
+	///Facehuggers overlay
+	var/mutable_appearance/hugger_overlays_icon
 
 // ***************************************
 // *********** Life overrides
 // ***************************************
-/mob/living/carbon/xenomorph/carrier/Stat()
+/mob/living/carbon/xenomorph/carrier/Initialize(mapload)
 	. = ..()
+	hugger_overlays_icon = mutable_appearance('icons/Xeno/2x2_Xenos.dmi',"empty")
 
+/mob/living/carbon/xenomorph/carrier/Stat()
+	. = ..()
 	if(statpanel("Game"))
 		stat("Stored Huggers:", "[huggers] / [xeno_caste.huggers_max]")
+
+/mob/living/carbon/xenomorph/carrier/update_icons()
+	. = ..()
+	update_overlays()
+
+/mob/living/carbon/xenomorph/carrier/update_overlays()
+	. = ..()
```

Comment:
```suggestion
```
update_overlays is part of the standard icon update procs and shouldnt be called directly, you can just put this in normally

Reply:
Done